packages/zowe-explorer/src/webviews/src/table-view/ActionsBar.tsx

```diff
@@ -1,66 +1,174 @@
+                alignItems: "center",
+                height: "1.25rem",
+                marginLeft: "-1.25rem",
+                overflow: "hidden",
+              }}
+            >
+              <span style={{ paddingRight: visibleColumns.includes(col) ? "1em" : "2.15em", marginTop: "3px" }}>
+                {visibleColumns.includes(col) ? <span className="codicon codicon-check"></span> : null}
+              </span>
+              {col}
+            </div>
+          )}
+        </MenuItem>
+      ));
+
   return (
     <div
       style={{
-        height: "2.5em",
+        height: "3em",
         display: "flex",
         alignItems: "center",
-        borderRadius: "var(--ag-wrapper-border-radius)",
         border: "1px solid var(--vscode-editorWidget-border)",
+        borderTopLeftRadius: "var(--ag-wrapper-border-radius)",
+        borderTopRightRadius: "var(--ag-wrapper-border-radius)",
         justifyContent: "space-between",
         backgroundColor: "var(--vscode-keybindingTable-headerBackground)",
         color: "var(--vscode-foreground) !important",
         padding: "0 0.25em",
-        marginBottom: "3px",
+        marginBottom: "-1px",
       }}
     >
-      <h5 style={{ marginLeft: "0.25em" }}>
-        {itemCount === 0 ? "No" : itemCount} item{itemCount === 1 ? "" : "s"} selected
-      </h5>
-      <span style={{ marginBottom: "0.25em" }}>
+      <h3 style={{ marginLeft: "0.25em" }}>
+        {title} ({itemCount})
+      </h3>
+      <span style={{ display: "flex", alignItems: "center", marginBottom: "0.25em" }}>
+        <p style={{ fontSize: "0.9em", paddingTop: "2px", marginRight: "0.75em" }}>
+          {selectionCount === 0 ? "No" : selectionCount} item{selectionCount > 1 || selectionCount === 0 ? "s" : ""} selected
+        </p>
         {actions
           .filter((action) => (itemCount > 1 ? action.callback.typ === "multi-row" : action.callback.typ.endsWith("row")))
-          .map((action, i) => (
-            <VSCodeButton
-              key={`${action.command}-action-bar-${i}`}
-              type={action.type}
-              style={{ height: "1.5em", fontWeight: "bold", marginRight: "0.25em" }}
-              onClick={(_event: any) => {
-                const selectedRows = (gridRef.current.api as GridApi).getSelectedNodes();
-                if (selectedRows.length === 0) {
-                  return;
-                }
+          .map((action, i) => {
+            // Wrap function to properly handle named parameters
+            const selectedRows = gridRef.current?.api?.getSelectedRows() ?? 0;
+            const cond = action.condition ? new Function(wrapFn(action.condition)) : undefined;
+            // Invoke the wrapped function once to get the built function, then invoke it again with the parameters
+            let shouldDisable = selectionCount === 0;
+            if (cond != null) {
+              shouldDisable ||= !cond()(action.callback.typ === "multi-row" ? selectedRows : selectedRows[0]);
```

Comment:
I might be misinterpreting something here but... is this a typo?
`action.callback.typ -> action.callback.type`?

Reply:
`type` is a reserved keyword so I used the shorthand as a way to define a "type" property 😅
This could probably be renamed in the future but since it the framework was already merged with `typ` as the property name, I'm hesitant to make the change